lib/runtime-core/src/backing.rs

```diff
@@ -143,17 +145,49 @@ impl LocalBacking {
                 }
             } as usize;
 
+            // Validate data size fits
             match init.memory_index.local_or_import(&module.info) {
                 LocalOrImport::Local(local_memory_index) => {
                     let memory_desc = module.info.memories[local_memory_index];
                     let data_top = init_base + init.data.len();
-
                     if memory_desc.minimum.bytes().0 < data_top || data_top < init_base {
                         return Err(vec![LinkError::Generic {
                             message: "data segment does not fit".to_string(),
                         }]);
                     }
+                }
+                LocalOrImport::Import(imported_memory_index) => {
+                    // Write the initialization data to the memory that
+                    // we think the imported memory is.
+                    unsafe {
```

Comment:
Can we move the unsafe only for the local_memory retrieval?

```rust
let local_memory = unsafe { &*imports.vm_memories[imported_memory_index] };
 ```

Reply:
Done, I also minimized the other `unsafe` block at 38a8a0eb01d2802f7a5899e7276653d075f21a2d